searchhub-fusion-plugins/src/main/java/com/lucidworks/apollo/pipeline/index/stages/searchhub/mail/MBoxParsingStage.java

```diff
@@ -37,7 +37,8 @@ public void process(PipelineDocument doc, Context context, StageOutput<PipelineD
     logger.info("running MBoxParsingStage on " + doc.getId() + " (which has " + doc.getAllFieldNames().size() + " fields )");
 
     // TODO: should be cached for better throughput
-    MimeMailParser parser = new MimeMailParser();
+    MBoxParsingStageConfig config = getConfiguration();
+    MimeMailParser parser = new MimeMailParser(config.getIdPattern(), config.getBotEmailPatterns());
```

Comment:
so wait, this seems thread-unsafe, right?  We're passing in the same patterns, maybe we should just be passing the strings in, and compiling them in each MimeMailParser?